packages/hint/docs/contributor-guide/getting-started/architecture.md

```diff
@@ -1,78 +1,75 @@
+
+[![webhint's architecture](images/sequence.svg)](images/sequence.svg)
+
+1. The `CLI` creates a new `HintConfig` object based on the user's `.hintrc`
+   file. This process validates that the hints are configured correctly, etc. If
+   the there is no `.hintrc` file, `webhint` will use a [default
+   configuration][default configuration].
+1. The `CLI` then passes this `HintConfig` to the `resource-loader` that will
+   return a `HintResources` object that contains the `Constructors` of the
+   configured resources. `resource-loader` checks at the same time if the
+   resources actually exist, are compatible with the current `webhint` version,
+   etc. If there are missing dependencies, the names will be in the `missing`
+   property of the result. For incompatibilities, it will be the `incompatible`.
+1. If everything goes well, a new `Engine` object is created using the previous
+   `HintConfig` and `HintResources`, then `CLI` calls its `executeOn` method.
+1. `Engine` then calls the `collect` method (`async`) of the configured
+   `connector`.
+1. The `connector` will navigate to the `URL`, traverse the HTML, and send
+   `event`s related to this and the loaded resources.
 1. If a `parser` has subscribed to one of the emitted `event`s, it will parse
    that resource and `emit` new `event`s.
 1. `hint`s can subscribe to any `event` no matter where it's comming from
    (`connector` or `parser`). If they find an issue, it will be reported via the
    `report` method.
-1. Once `webhint` has finished analyzing the website, the results are returned
-   to `CLI` that will print them using the configured `formatter`(s).
+1. Once `collect` returns, the results are passsed then to `CLI` that will call
+   the `format` method of all the configured `formatters`s.
 
-Any developer can create their own `hint`s, `connector`s, `parser`s
-and/or `formatter`s, and use them without having to do a pull request to
-the main project and distribute them as [`npm`][npm] packages.
+Any developer can create their own `hint`s, `connector`s, `parser`s and/or
+`formatter`s, and use them without having to do a pull request to the main
+project and distribute them as [`npm`][npm] packages.
 
-## CLI Flow
+## Resource loading
 
-You can learn more about how `webhint` loads the configuration and resources
-in this diagram:
+The resource loading of `webhint` for the CLI has many steps. The following is
+the most up-to-date diagram of the interaction:
 
 [![webhint's flow diagram](images/cli.svg)](images/cli.svg)
 
 <!-- Link labels: -->
 
-[default configuration]: ../../user-guide/#default-configuration
-[events]: ./events/
-[new connector]: ../how-to/connector/
-[new formatter]:../how-to/formatter/
-[new parser]: ../how-to/parser/
-[new hint]: ../how-to/hint/
-[npm]: https://www.npmjs.com/
-[typescript]: https://www.typescriptlang.org/
+[default configuration]: ../../user-guide/#default-configuration [events]:
```

Comment:
This needs to be fixed.

Reply:
Done, I'm using a new extension that automatically wraps content at 80 characters and looks like it messes link labels 😔